account_invoice_currency_by_partner/models/account_move.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2025 Tecnativa - Eduardo Ezerouali
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.depends("partner_id.vendor_currency_id", "partner_id.customer_currency_id")
```

Comment:
Watch out! You shouldn't put a dependency that affects previous records. Imagine all the past invoices being changed because you now change the currency in the partner...

```suggestion
    @api.depends("partner_id")
```

Reply:
change done.